lib/grant.js

```diff
@@ -449,7 +452,24 @@ function saveRefreshToken (done) {
 }
 
 /**
- * Create an access token and save it with the model
+ * Pass the scope string to the model for saving
+ *
+ * @param  {Function} done
+ * @this   OAuth
+ */
+function saveScope (done) {
+  var scope = this.scope || this.req.body.scope;
+
+  var self = this;
+  this.model.saveScope(scope, self.accessToken, function (err, acceptedScope) {
+    if (err) return done(error('server_error', false, err));
```

Comment:
Nice solution with the `allowed` boolean there. Regardless of whether or not we go with the separate scope middleware, that should definitely be worked into the scope check. :+1: